cache/cache.go

```diff
@@ -144,6 +144,9 @@ func (c *cache) Remove(keys ...string) {
 }
 
 func (c *cache) Size() uint64 {
+	c.Lock()
+	defer c.Unlock()
```

Comment:
A code comment saying what's in the PR description would be really helpful, I think.